TreeMaker/python/maker.py

```diff
@@ -56,12 +59,26 @@ def __init__(self,parameters):
         self.getParamDefault("applybaseline", False)
         self.getParamDefault("saveMinimalGenParticles", True)
 
+        # branch control options
+        self.getParamDefault("includeBranches", "")
+        self.getParamDefault("excludeBranches", "")
+        self.getParamDefault("exactBranches", False)
+        if len(self.includeBranches)>0 and len(self.excludeBranches)>0:
+            raise ValueError("includeBranches and excludeBranches are exclusive options; pick one!")

```

Comment:
Are these really exclusive if using regex? Can't you use a simpler regex to exclude most branches of a pattern, but include just a few? That is opposed to writing a really long regex with the inclusions excluded from what the regex matches.

Reply:
But the order becomes ambiguous: the user would need to specify whether include or exclude comes first. Then one could imagine even more complicated use cases, but I don't see a motivation to support them right now.